File: src/Stack.java
/**
 * Created by doramedgyasszay on 2017. 04. 19..
 */
public class Stack {

    Node top;

    public void push(Object object){
        Node node = new Node(object);
        if(top == null) {
            top = node;
        } else {
            node.setPrev(top);
            top.setNext(node);
            top = node;
        }
    }

    public Object pop(){
        if(top == null){
            return null;
        } else {
            Object result = top.getItem();
            top = top.getPrev();
            if(top != null){
                top.setNext(null);
            }
            return result;
        }
    }

    public static void main(String[] args) {
        Stack stack = new Stack();
        stack.push(5);
        stack.push(52);
        stack.push(511);
        System.out.println(stack.pop());
        System.out.println(stack.pop());
        System.out.println(stack.pop());
        System.out.println(stack.pop());

    }
}

File: src/js/Stack.js
/**
 * Created by doramedgyasszay on 2017. 05. 03..
 */

var Stack = function () {
    this.top = null;
    this.count = null;
};

var Node = function(data){
    this.data = data;
    this.next = null;
    this.prev = null;
};


Stack.prototype.push = function (data) {
    var node = new Node(data);
    if (this.top === null) {
        this.top = node;
    } else {
        node.prev = this.top;
        this.top.next = node;
        this.top = node;

    }
    this.count++;
};

Stack.prototype.pop = function () {
        if (this.top === null){
            return null;
        }
        else {
            var out = this.top;
            this.top = this.top.prev;
            if(this.top != null){
                this.top.next = null;
            }
            this.count--;
            return out;
        }
    };

Stack.prototype.displayAll = function () {
        if(this.top === null){
            return null;
        } else {
            var array = new Array();
            var arrLength = this.count;
            var current = this.top;
            for(var i = 0; i < arrLength; i++){
               array[i] = current.data;
                current = current.prev;
            }
            return array;
        }
};


var stack = new Stack();
stack.push(34);
stack.push(3224);
stack.push(11);
console.log("top element is: " + stack.top.data);
console.log("Size is: " + stack.count);
console.log("Display all: " + stack.displayAll());
stack.pop();
console.log("top element is: " + stack.top.data);


    
